### DISCUSSION TITLE
Can the Tuya wifi energy consumption meter still be used?

### DISCUSSION THREAD

**Discussion by ljeffyl** [make-all/tuya-local#1624] [Q&A]
![image](https://github.com/make-all/tuya-local/assets/96488808/0742c291-9095-444e-8335-596237d3cee7)
I bought four V-WIFI-DL02-ES. I found that Tuya has local support, but cannot add HA (I added it using Tuya APP, not Smart Life. I don’t know if it has anything to do with this. I will try it later)

During configuration, "Unable to connect to your device details. This may be an intermittent issue or it may be an error."
I also tried the auto version
![未命名](https://github.com/make-all/tuya-local/assets/96488808/170d7f53-020b-4002-bd1b-68bd0e40684e)

The following messages will appear in the log in order:
<class 'OSError'>: [Errno 98] The address used when initializing device eb153b53cf9d273cf3src1

Connection test failed, <class 'OSError'> [Errno 98] The address is in use

An error occurred while executing the job: Exception in callback _SelectorDatagramTransport._read_ready()
Traceback (most recent call last):
           File "/config/custom_components/localtuya/discovery.py", line 65, datagram_received
             data = decrypt_udp(data)
                    ^^^^^^^^^^^^^^^^^
           File "/config/custom_components/localtuya/discovery.py", line 30, decrypt_udp
             return_unpad(decryptor.update(message)+decryptor.finalize()).decode()
                                                       ^^^^^^^^^^^^^^^^^^^
           File "/usr/local/lib/python3.11/site-packages/cryptography/hazmat/primitives/ciphers/base.py", line 186, finalized
             data = self._ctx.finalize()
                    ^^^^^^^^^^^^^^^^^^^
           File "/usr/local/lib/python3.11/site-packages/cryptography/hazmat/backends/openssl/ciphers.py", line 222, finalized
             raises valueError(
ValueError: The data length provided is not a multiple of the block length.

In the process of handling the above exception, another exception occurred:

Traceback (most recent call last):
           File "/usr/local/lib/python3.11/asyncio/events.py", line 80, in _run
             self._context.run(self._callback, *self._args)
           File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 1163, in _read_ready
             self._protocol.datagram_received(data, address)
           File "/config/custom_components/localtuya/discovery.py", line 67, datagram_received
             data = data.decode()
                    ^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec cannot decode byte 0xb9 in position 0: invalid starting byte

**Comment by make-all** [comment]
Are you trying to use Auto IP address detection, with localtuya also installed?

**Reply by ljeffyl** [reply]
I have tried both, automatic or specified, and it is installed for me. Will this have any impact?

**Reply by make-all** [reply]
Did you try different protocol versions?  In the screenshot the protocol is set to 3.3 and the log is showing decryption errors toward the bottom, but they are coming from localtuya, not from tuya-local.  If you want to try this integration you must remove it from your localtuya configuration first.

**Reply by ljeffyl** [reply]
I have tried all versions, but that was when localtuya existed but no device was configured. After just deleting localtuya, I can see all the sensors with version 3.5, so will there be a conflict between the two?

**Reply by ljeffyl** [reply]
Thank you very much for your development and answers

**Reply by ljeffyl** [reply]
I would like to ask how often these sensors are asked? Can they be adjusted by themselves?

**Reply by make-all** [reply]
There is a conflict between localtuya and this integration in the following situations:
 - using Auto IP address detection
 - having the same device configured in both

If you manually configure your devices with fixed IP, and the devices you configure in tuya-local are not configured in localtuya, then they should be able to work side by side.

When connected normally with a persistent connection, the sensor controls how often it updates. If you tick "polled updates", then Home Assistant controls how often the polling happens.

**Reply by ljeffyl** [reply]
I have a problem now. Unable to connect to other identical devices. It will say "Sorry, this device is not supported."

![image](https://github.com/make-all/tuya-local/assets/96488808/7a44dd36-69e6-4bc3-858b-a7e033398449)
